extended-statistics.md

```diff
@@ -0,0 +1,114 @@
+
+## The type of the Extended Statistics
+
+### Correlation
+
+This is the only supported type of extended statistics. The registration SQL is like the following:
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE t ADD STATS_EXTENDED s1 correlation(col1, col2);
+```
+
+When we run the `ANALYZE` after the registration, TiDB will calculate the [Pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) of the `col` and `col2` of the table `t` and write the record into the table `mysql.stats_extended`.
+
+It's used to improve TiDB's index selection for the following scenario:
+
+For a table `t` described below：
+
+{{< copyable "sql" >}}
+
+```sql
+CREATE TABLE t(col1 INT, col2 INT, KEY(col1), KEY(col2));
+```
+
+Suppose that the `col1` and `col2` of the table `t` both obey monotonically increasing constraints in row order, i.e., the values of `col1` and `col2` are strictly correlated in order (correlation value of 1):
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t WHERE col1 > 1 ORDER BY col2 LIMIT 1;
+```
+
+For the above query, the optimizer has two choices to access the table `t`: one uses the index on `col1` to access the table and then sorts the result by `col2` to calculate the `Top-1`. Another is that access the table by index on `col2` to meet the first row that satisfies `col1 > 1`. The latter's cost mainly depends on how many rows are filtered out when we scan the table in `col2`'s order. Usually, the optimizer can only suppose that `col1` and `col2` are independent, leading to a significant estimation error.
+
+After the TiDB has the extended statistics for correlation, the optimizer can estimate how many rows we need to scan more precisely. Since the `col1` and `col2` are strictly correlated in order, the optimizer will equivalently translate the row count estimate for option two above into:
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t WHERE col1 <= 1 OR col1 IS NULL;
+```
+
+The above estimation plus one will be the final estimation for the condition. This way, we don't need to use the independent assumption to get a significant estimation error.
+The optimizer will use the independent assumption if the correlation factor is less than the system variable `tidb_opt_correlation_threshold`. But it will increase the estimation heuristically. The larger the system variable `tidb_opt_correlation_exp_factor` is, the larger the estimation result is. The larger the absolute value of the correlation factor is, the larger the estimation result is.
+
+## The collection of the Extended Statistics
+After registration, the TiDB will collect the extended statistic after triggering the ANALYZE command manually or automatically. TiDB will not collect the extended statistics if we only collect the indexes' statistics. It also will not collect it if it's `ANALYZE INCREMENTAL` or the `tidb_enable_fast_analyze` is true.
```

Comment:
The description is a little redundant.  I would recommend the follows:

After registration, TiDB collects the extended statistic with the ANALYZE command manually or automatically, except below scenarios:
- Statistics collection on indexes only
- Statistics collection with `ANALYZE INCREMENTAL` command. 
- Statistics collection with variable `tidb_enable_fast_analyze=true`